ross/api_report

```diff
@@ -0,0 +1,538 @@
+                        left=[SMspeed],
+                        right=[wn[i]],
+                        tag1=[wn[i]],
+                        tag2=[SMspeed],
+                    )
+                )
+
+                mag_plot.quad(
+                    top="top",
+                    bottom="bottom",
+                    left="left",
+                    right="right",
+                    source=source,
+                    line_color=bokeh_colors[8],
+                    line_width=0.8,
+                    fill_alpha=0.2,
+                    fill_color=bokeh_colors[8],
+                    legend="Separation Margin",
+                    name="SM2",
+                )
+                hover = HoverTool(names=["SM2"])
+                hover.tooltips = [
+                    ("Critical Speed :", "@tag1"),
+                    ("Speed at 0.707 x peak amplitude :", "@tag2"),
+                ]
+                mag_plot.add_tools(hover)
+
+            else:
+                SM = None
+                SM_ref = None
+                SMspeed = None
+
+            AF_table.append(AF)
+            SM_table.append(SM)
+            SM_ref_table.append(SM_ref)
+
+        # amplitude limit in micrometers (A1) - API684 - SP6.8.2.11
+        A1 = 25.4 * np.sqrt(12000 / (30 * maxspeed / np.pi))
+
+        # amplitude from mode shape analysis
+        Amax = max(mag[dof])
+
+        # Correction Factor (CF) - API684 - SP6.8.2.11
+        CF = max(A1 / Amax, 0.5)
+
+        mag_plot.quad(
+            top=max(mag[dof]),
+            bottom=0,
+            left=minspeed,
+            right=maxspeed,
+            line_color="green",
+            line_width=0.8,
+            fill_alpha=0.2,
+            fill_color="green",
+            legend="Operation Speed Range",
+        )
+
+        source = ColumnDataSource(dict(x=freq_range, y=mag[dof]))
+        mag_plot.line(
+            x="x",
+            y="y",
+            source=source,
+            line_color=bokeh_colors[0],
+            line_alpha=1.0,
+            line_width=3,
+        )
+
+        return mag_plot
+
+    def mode_shape(self, mode):
+        """Evaluates the mode shapes for the rotor.
+
+        This analysis presents the vibration mode for each critical speed.
+        The importance is to locate the critical node, where the displacement
+        is the greatest, then apply loads for unbalance response (stability
+        level 1)
+
+        Parameters
+        ----------
+        mode : int
+            the n'th vibration mode
+
+        Returns
+        -------
+        node_min, node_max : list
+            List with nodes where the largest absolute displacements occur
+
+        Example
+        -------
+        >>> rotor = rotor_example()
+        >>> report = Report(rotor=rotor)
+        >>> report.mode_shpae(mode=0) # doctest: +ELLIPSIS
+        [...]
+        """
+        modes = (self.rotor.evectors[: self.rotor.ndof]),
+        evec0 = modes[0][:, mode]
```

Comment:
This code repeats what we have in `ModeShapeResults`. 
Instead of repeating all the code here, we should change `ModeShapeResults` so that the information we need is available for us to use here.

Reply:
I repeated the code because I wanted to avoid making modifications to the main program. But following your suggestions, I change what ModeShapesResults returns.

Reply:
I repeated the code because I wanted to avoid making modifications to the main program. But following your suggestions, I change what ModeShapesResults returns.